src/cloudant/index.py

```diff
@@ -262,3 +262,91 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes.
+    """
+    def __init__(
+            self,
+            ddoc,
+            search_index_name=None,
+            search_index=None
```

Comment:
I think this has 1 level of indentation too much.  Remove 4 spaces.  I see that the `View.__init__()` has that as well.  Seems like that's where this originated.  It is unfortunate that pylint decided that it was not interested in catching that.  Anyway, can you create a separate PR to fix the `View.__init__()`?